tests-e2e/core/snapshotenvironmentbinding_test.go

```diff
@@ -687,26 +689,32 @@ var _ = Describe("SnapshotEnvironmentBinding Reconciler E2E tests", func() {
 			By("Verify whether `gitopsDeployment.ObjectMeta.Labels` is updated with ASEB labels")
 			err = k8s.Get(&gitopsDeployment, k8sClient)
 			Expect(err).To(BeNil())
-			Expect(gitopsDeployment).To(gitopsDeplFixture.HaveLabel("appstudio.openshift.io", "testing-update"))
+			Eventually(gitopsDeployment, "2m", "10s").Should(gitopsDeplFixture.HaveLabel("appstudio.openshift.io", "testing-update"))
 
 			By("Remove ASEB label `appstudio.openshift.io` label and verify whether it is removed from gitopsDeployment label")
 			delete(binding.ObjectMeta.Labels, "appstudio.openshift.io")
-			err = k8s.Update(&binding, k8sClient)
+			err = k8s.UntilSuccess(k8sClient, func(k8sClient client.Client) error {
+				// Retrieve the latest version of the SnapshotEnvironmentBinding resource
+				err := k8s.Get(&binding, k8sClient)
```

Comment:
I think this is a no-op: it will retrieve the Binding, and then call Update using that exact same (unchanged) value.

Reply:
Adding the delete call inside the `UntilSuccess` closure